### ISSUE TITLE
Metadata not fetched for tiledMapLayer with subdomains configured 

### ISSUE THREAD

**Issue by Biboba** [Esri/esri-leaflet#991]
* Browser and version: 
Chromium Version 59.0.3071.109 (Build de développement) Built on Ubuntu , running on Ubuntu 16.04 (64 bits)

* Version of Leaflet (`L.version`): 1.2.0

* Version of esri Leaflet (`L.esri.VERSION`): 2.1.0

Steps to reproduce the error:

When using a L.esri.tiledMapLayer configured with subdomains (eg: https://{s}.arcgisonline.com/arcgis/rest/services/World_Topo_Map/MapServer, with ['server', 'services'] as subdomains),  
service metadata are not fetched as the url contains '{s}' notation throwing an "net::ERR_NAME_NOT_RESOLVED" error on Chromium Version 59.0.3071.109. 

Not critical though as the tiles get properly displayed.

On the following jsbin, I am using an arcgis online service as an example but I know I can use the 'basemapLayer' method which do not send this request.

http://jsbin.com/soropohemi/edit?html,output

**Comment by jgravois** [comment]
thats a good catch. thank you for the thorough report! just to reiterate (and clarify)

### the bug

```js
var map = L.map("map").setView([37.75, -122.23], 10);
L.esri.tiledMapLayer({
  url: 'https://{s}.arcgisonline.com/arcgis/rest/services/World_Topo_Map/MapServer',
  subdomains: ['server', 'services']
}).addTo(map);
```

the code above *should* cause the copyright text printed [here](https://services.arcgisonline.com/arcgis/rest/services/World_Topo_Map/MapServer) to display in Leaflet's attribution control.
![screenshot 2017-08-12 21 50 09](https://user-images.githubusercontent.com/3011734/29246237-2c3a9a52-7fa8-11e7-9bbc-45e172a8a2cc.png)

this is what currently displays

![screenshot 2017-08-12 21 50 25](https://user-images.githubusercontent.com/3011734/29246240-3ea1646e-7fa8-11e7-9ce3-248e6960f81c.png)

if anyone is interested in helping patch this bug (i'm looking at you @daraobeirne 😺 ), its a pretty **fun** one!

Leaflet itself has a convention for allowing developers to specify multiple subdomains for tile servers and substitutes them at runtime when fetching tiles (this is called [domain sharding](https://blog.stackpath.com/glossary/domain-sharding/) and it will remain a good optimization to workaround limits of six concurrent requests to a domain until HTTP/2 becomes more prevalent).

```js
/* leaflet employs logic under the hood
to substitute hard values for `{s}`, `{z}`, `{y}` and `{x}` 
when it comes time to fetch each individual tile.
*/
L.tileLayer({
  url: 'https://{s}.arcgisonline.com/arcgis/rest/services/World_Topo_Map/MapServer/tile/{z}/{y}/{x}',
  subdomains: ['server', 'services']
}).addTo(map);
```

you can dig in and see how it works in [`L.tileLayer.getTileUrl()`](https://github.com/Leaflet/Leaflet/blob/1491de5acde38d7e3ee45073845dfcbbabe8a334/src/layer/tile/TileLayer.js#L161-L178)

here in Esri Leaflet-land, we've *always* taken advantage of Leaflet's support for subdomains for Esri's own hosted tiles (`L.esri.basemapLayer`) and never noticed the fact that copyright information isn't fetched when the layer first loads because we have a much *niftier* technique for displaying attribution dynamically as users pan and zoom that gives credit to whichever of our dozens and dozens of community basemap data providers is relevant at the moment.

### how do we fix it?

@Biboba has already pointed us most of the way there. we need to make sure that when an `L.esri.tiledMapLayer` is first added to the map and metadata is fetched from the service for copyright text to display as attribution that we account for these subdomains if they are present.

the internal call to fetch service metadata happens [here](https://github.com/Esri/esri-leaflet/blob/f17663e481f6c738b0776f208f0ac14e835fa2ab/src/Layers/TiledMapLayer.js#L109)

this bugfix is **not** a softball, but i'm going to mark it as <kbd>First Timers Only</kbd> because its such a *great* learning opportunity and because developers have a straightforward workaround to define static attribution in their own code temporarily.

i'll accept a pull request from anyone that stretches a bit beyond their comfort zone to chip in and learn more and i'm happy to provide assistance if you get stuck along the way.

**Comment by jgravois** [comment]
resolved by @Biboba  in #993 :tada:
